pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -165,6 +153,16 @@ type ContainerSpec struct {
 	Limits          *ResourceRequirement `json:"limits,omitempty"`
 }
 
+// PodAttributesControlSpec is a spec of some general attributes of TiKV, TiDB and PD Pods
+type PodAttributesSpec struct {
+	Affinity           *corev1.Affinity           `json:"affinity,omitempty"`
+	NodeSelector       map[string]string          `json:"nodeSelector,omitempty"`
+	Tolerations        []corev1.Toleration        `json:"tolerations,omitempty"`
+	Annotations        map[string]string          `json:"annotations,omitempty"`
+	HostNetwork        bool                       `json:"hostNetwork,omitempty"`
+	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+}
```

Comment:
`go vet` will fail on this, because it does not allow using promoted fields in `PodAttributesSpec` outside of this package.

Reply:
I see... but `make check` passed in my local environment, is there no `go vet` check in `make check`?

Reply:
we can see the results of e2e test

Reply:
`go vet` does not fail on your PR, because there is no place to access fields in `PodAttributesSpec` in this way: `TiDBSpec.Affinity`.
I'd like to abstract these fields into one struct. What about using a shorter name, `PodSpec`. Because this struct name must be used to access internal fields, e.g `TiDBSpec.PodSpec.Affinity`.

Reply:
I've considered `PodSpec`, but IMHO it's a little ambiguous with the actual specification of **Pod**, which contains important fields like `containers`, `command`, etc.

Reply:
IMO it's fine, we already have `ContainerSpec`.

Reply:
Besides, access promoted fields is OK for go vet, the check is for struct literal:

```

type t1 struct {
	t1f1 int
	t1f2 int
}
type t2 struct {
	t2f1 int
	t2f2 int
	t1
}

var (
	_ = t2{t1f1: 600} // ERROR "cannot use promoted field t1.t1f1 in struct literal of type t2"
)

func main() {
    a := t2{
        t1: t1{t1f1: 2,t1f2: 3,},
        t2f1: 4,
        t2f2: 5,
    }
    _ = a.t1f1
    _ = a.t1f2

}
```

```
go vet
# _/Users/alei/repo/alei/vet [_/Users/alei/repo/alei/vet.test]
./main.go:14:9: cannot use promoted field t1.t1f1 in struct literal of type t2
```

Reply:
Having this, I think it's better to use a longer but more concise name

Reply:
you're right, it passes because there are no struct literals in test files.

Reply:
I I think it's better to remove the Pod prefix since it's already inside the podattributespec.

Reply:
Because we use `PodAttributesSpec` for composition, I think `PodSecurityContext` is more concise when access the field of PD/TiDB/TiKVSpec. e.g.: `TiDBSpec.PodSecurityContext`